src/components/depot/implementation.ts

```diff
@@ -0,0 +1,30 @@
+import { BlockCodec } from "multiformats/codecs/interface"
+import { CID } from "multiformats/cid"
+
+
+export type Implementation = {
+  // Get the data behind a CID
+  getBlock: (cid: CID) => Promise<Uint8Array>
+  getUnixFile: (cid: CID) => Promise<Uint8Array>
+  getUnixDirectory: (cid: CID) => Promise<DirectoryItem[]>
+
+  // Keep data around
+  putBlock: (data: Uint8Array, codec: BlockCodec<number, any>) => Promise<CID>
+  putChunked: (data: Uint8Array) => Promise<PutResult>
```

Comment:
I should probably explain here that chunked means a collection of various blocks grouped together by a DAG-
PB node.